### PR TITLE
Implement `Hbar` (PR)

### PR DESCRIPTION
**Pull request by deyanzz**
**Description**:

This PR adds the missing functionality for the `Hbar` class.

**Related issue(s)**:

**Fixes**: https://github.com/hashgraph/hedera-sdk-cpp/issues/552

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/main/include/Hbar.h [hunk 1]
```diff
@@ -32,6 +33,10 @@ namespace Hedera
  * Implemented as a wrapper class to force handling of units. Direct interfacing with Hedera accepts amounts in tinybars
  * however the nominal unit is hbar.
  */
+
+// Define the pattern
+const std::regex FROM_STRING_PATTERN("^((?:\\+|\\-)?\\d+(?:\\.\\d+)?)(\\ (tℏ|μℏ|mℏ|ℏ|kℏ|Mℏ|Gℏ))?$");
```
Comment (rwalworth) [inline sdk/main/include/Hbar.h:38]: I don't think this needs to be exposed in the header file, you could just throw it in the source file within an anonymous namespace

**Review by rwalworth — APPROVED** [review]
LGTM

File: sdk/main/include/Hbar.h (hunk 1 above)
Comment (deyanzz) [reply sdk/main/include/Hbar.h]: Fixed in af94ce7e0647ed4502fa8593960ba403160dea5d

**Review by gsstoykov — APPROVED** [review]
LGTM.

File: sdk/main/src/Hbar.cc [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                                "\" was not correctly formatted");
+  }
+
+  // Extract parts from the match
+  std::string amountStr = match[1].str();
+  std::string unitStr = match[3].str();
+
+  // Convert amount string to BigDecimal
+  std::istringstream amountStream(amountStr);
+  int64_t amount;
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/main/src/Hbar.cc:64]: Thanks. It's fixed in 5533406b7eae36928970ecb143866629c67ea782.

**Pull request merged**